### PR TITLE
Sphere diffusion unit test

### PR DESCRIPTION
**Pull request by valeriabarra**
Add a Poisson's problem on the sphere unit test, to test the spatial Laplacian only.

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -1,26 +1,28 @@
 using Base: operator_associativity
 
-include("recursive.jl")
```
Comment (jakebolewski) [inline test/runtests.jl:3]: I would revert this change as the original ordering  in runtests is useful for debugging because it better captures the rough dependency order of functionality for the tests

**Review by jakebolewski — CHANGES_REQUESTED** [review]

File: test/runtests.jl (hunk 1 above)
Comment (valeriabarra) [reply test/runtests.jl:3]: Oh, I thought the alphabetical order was just better organized for people to look for existing tests and insert new ones. Also, the other day I found a test that was included twice (`diffusion2d.jl`) and I thought that if they were alphabetized it would be clearer to avoid double entries.

Comment (jakebolewski) [reply test/runtests.jl:3]: it might be good to add a comment about why the ordering is the way it is for the future

File: test/sphere_diffusion.jl [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+using StaticArrays, IntervalSets, LinearAlgebra
+
+@testset "Scalar Poisson problem - ∇⋅∇ = f on the cubed-sphere" begin
+    # Poisson equation on a sphere
+    # - ∇⋅(∇ u(ϕ, λ)) = f(ϕ, λ)
+
+    # With:
+    # ϕ = asin(z / R) # latitude (R radius of sphere)
+    # λ = atan2(y, x) # longitude ∈ [-π, π]
+    # True solution (eigenfunction): u(ϕ, λ) = sin(λ) * cos(ϕ)
```
[the remark this answers is not in the document]
Comment (valeriabarra) [reply test/sphere_diffusion.jl:16]: Yes, I know spherical harmonics are, but this is a test case that it's used (see https://www2.cisl.ucar.edu/sites/default/files/levy_poisson_0.pdf and other Finite Element packages that include this as a test)

File: test/runtests.jl (hunk 1 above)
Comment (valeriabarra) [reply test/runtests.jl:3]: Done.

**Comment by jiahe23** [comment]
> > > @jiahe23 do you think this looks good? If yes, can approve it?
> > 
> > 
> > I think for sphere tests, we should use spherical harmonics as eigenfunctions as in [here](https://github.com/CliMA/ClimaCore.jl/blob/d6e480dbce58ff420820de4e511619afdf7ed9bb/test/hyperdiffusion_sphere.jl#L15)
> 
> I don't think it's really necessary in this case. I left a reference where this test was formalized and it's also used in other Finite Element packages. It works and I also checked the spectral superconvergence properties.

Sounds good!

**Review by jiahe23 — APPROVED** [review]

**Comment by jakebolewski** [comment]
it would be good to squash commits going forward for PR's like this before merging
